### PR TITLE
Start of Building on Windows doc

### PR DESCRIPTION
**Pull request by smurawski**
Continuing on the Windows build saga ( #1192 )

Adding a BUILDING_WINDOWS doc to help others trying to get a development environment going for Windows Habitat dev work.

Signed-off-by: Steven Murawski <steven.murawski@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
@smurawski, thanks for your PR! By analyzing the annotation information on this pull request, we identified @adamhjk to be a potential reviewer

File: docs/BUILDING_WINDOWS.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Building Habitat from source
+
+*NOTE:* This document is likely to change a lot over the coming months.
+
+*NOTE:* For Rust editing, I'd recommend Visual Studio Code and have [a blog post on how to set up the best Rust editing experience](http://stevenmurawski.com/powershell/2016/08/rustify).
```
Comment (reset) [inline docs/BUILDING_WINDOWS.md]: These notes are *very* helpful. A lot of people have their own tool bench but it doesn't have great Rust support just yet

Comment (smurawski) [inline docs/BUILDING_WINDOWS.md]: Thanks @reset 

Turns out that VS Code has pretty good rust support (not just on Windows), but I didn't want to make that assumption for folks in the doc.

**Comment by fnichol** [comment]
Great start and I'm happy for docs beach head!

gif-keyboard-15357566727221214362

**Pull request merged**